cpp/arcticdb/storage/s3/s3_api.cpp

```diff
@@ -26,10 +28,20 @@ S3ApiInstance::S3ApiInstance(Aws::Utils::Logging::LogLevel log_level) :
     }
     ARCTICDB_RUNTIME_DEBUG(log::storage(), "Begin initializing AWS API");
     Aws::InitAPI(options_);
-    // TODO: this is potentially a workaround for https://github.com/aws/aws-sdk-cpp/issues/1410
-    //       disabling it on windows.
-#ifndef _WIN32
-    setenv("AWS_EC2_METADATA_DISABLED", "true", true);
```

Comment:
What does this simpler workaround cost us? What is the downside of always disabling this "metadata" feature?

Reply:
This will affect every library running in the same and child processes.

For example, it will be pretty bad if the user runs the code inside AWS, loaded arcticdb and suddenly lost access to all AWS services discovered via the metadata.